benches/search_comparison.rs

```diff
@@ -165,36 +165,32 @@ fn search_bench_case<const MAX: usize, T, Coll>(
     setup_fun: impl Fn(Vec<T>) -> Coll,
     search_fun: impl Fn(&Coll, T) -> Option<&T>,
     group: &mut BenchmarkGroup<WallTime>,
-    i: &usize,
+    size: &usize,
     duplicates: bool,
 ) where
-    T: TryFrom<usize> + Ord + std::ops::Rem<Output = T> + num_traits::ops::wrapping::WrappingMul,
+    T: TryFrom<usize> + Ord + Clone,
     <T as TryFrom<usize>>::Error: core::fmt::Debug,
 {
-    group.bench_with_input(BenchmarkId::new(name, i), i, |b, i| {
-        let size = *i;
+    group.bench_with_input(BenchmarkId::new(name, size), size, |b, &size| {
+        // increasing sequence of even numbers, bounded by MAX
+        let iter = (0usize..)
+            .map(|i| std::cmp::min(i * 2, MAX))
+            .map(|i| T::try_from(i).unwrap());
```

Comment:
Something still feels off about this. In particular, once `i * 2` hits `MAX`, then we'll just generate `MAX` over and over again, which also feels wrong?

Reply:
Which makes me think this should also be `% MAX` (like you've changed the iterator to). Although that feels wrong too, because then you might end up with duplicates even `if !duplicates`. So I wonder if this needs to be an assert.

Reply:
IMO it should be `% MAX`. Although it can generate some amount of duplicates in case of `!duplicates` this amount should be negligible. At least when `MAX` much greater than number of values generated. It obviously will not work in case of `u8`, but in this case no solution exists :)

Reply:
Replaced with `% MAX`